src/postgres/commands/checkAuthentication.ts

```diff
@@ -24,10 +29,22 @@ export async function checkAuthentication(context: IActionContext, treeItem: Pos
                 // Need to configure firewall only for Azure Subscritption accounts
             } else if (treeItem.parent.resourceGroup && parsedError.errorType === firewallNotConfiguredErrorType) {
                 await configurePostgresFirewall(context, treeItem.parent);
+            } else if (treeItem.parent.resourceGroup && parsedError.errorType === 'ETIMEDOUT' && !(await isFirewallRuleSet(treeItem.parent))) {
+                await configurePostgresFirewall(context, treeItem.parent);
             } else {
                 throw error;
             }
         }
     }
     return clientConfig;
 }
+
+export async function isFirewallRuleSet(treeItem: PostgresServerTreeItem): Promise<boolean> {
+    const serverType: PostgresServerType = nonNullProp(treeItem, 'serverType');
+    const client = createAbstractPostgresClient(serverType, treeItem.root);
+    client.firewallRules.listByServer
+    const result: FirewallRuleListResult = (await client.firewallRules.listByServer(nonNullProp(treeItem, 'resourceGroup'), nonNullProp(treeItem, 'azureName')))._response.parsedBody;
```

Comment:
how come this has to do `_response.parsedBody`?

Reply:
`listByServer()` returns `Models.FirewallRulesListByServerResponse` which is in this format. 

![image](https://user-images.githubusercontent.com/17537419/126241548-5c941580-d3cc-48c1-82fb-f8e6fd73b264.png)

Reply:
Look at the type again - the screenshot is cutoff on the right, but I'm guessing it's `FirewallRuleListResult & {`, meaning the object itself is the same type as `_response.parsedBody`